docs/nx-cloud/recipes/encryption.md

```diff
@@ -0,0 +1,24 @@
+# Enable End to End Encryption
+
+To turn on end to end encryption, specify an encryption key in one of two ways:
+
+- Set the `encryptionKey` property in `nx.json`
+- Set the `NX_CLOUD_ENCRYPTION_KEY` environment variable
+
+The key can be any string up to 32 characters long.
+
+Providing an encryption key tells Nx to encrypt task artifacts on your machine before they are sent to the remote cache. Then when cached results are downloaded to your machine they are decrypted before they are used. This ensures that even if someone gained access to the Nx Cloud servers, they wouldn't be able to view your task artifacts.
+
+## Metadata
+
+End to end encryption only applies to output files of a task. Terminal output and other metadata like the dependency graph and the time it takes a task to execute are still encrypted, but in such a way that Nx Cloud has access to them. This allows Nx Cloud to use that information to efficiently orchestrate tasks and to display your logs on nx.app.
```

Comment:
we could mention here explicitly that "all the artefacts NxCloud uses to replay a task for you are encrypted. That means that even if someone gets access to your NxCloud storage bucket, they will not be able to tamper with the files and terminal output that is restored when the task is replayed on your CI or developer's machines." (I think that makes it clear and aleviates the main security concern of companies)

We can then say that we "store an un-encrypted version of the terminal output separately that is accessible only to invited members of the workspace on the NxCloud webapp, so they can see why certain tasks failed. This un-encrypted output is only used in the browser, and not used when replaying the task."